google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java

```diff
@@ -132,11 +139,18 @@ public void computeDeployments(@NotNull ComputeDeploymentsCallback callback) {
   public void disconnect() {
     // kill any executing deployment actions
     synchronized (createdDeployments) {
-      for (CancellableRunnable runnable : createdDeployments) {
+      for (CancellableRunnable runnable : createdDeployments.values()) {
         runnable.cancel();
       }
       createdDeployments.clear();
     }
+
+    createdDeployments
+        .keys()
+        .forEach(
+            project ->
+                ProjectManager.getInstance()
```

Comment:
look at `ProjectManagerImpl#addProjectManagerListener`

looks like its registering the removal callback with the disposer on each add